test/Microsoft.OpenAPI.OData.Reader.Tests/Generator/OpenApiSchemaGeneratorTests.cs

```diff
@@ -80,6 +80,52 @@ public void CreateStructuredTypeSchemaThrowArgumentNullEnumType()
             Assert.Throws<ArgumentNullException>("structuredType", () => context.CreateStructuredTypeSchema(structuredType: null));
         }
 
+        [Fact]
+        public void CreateStructuredTypeSchemaWithDiscriminatorValueEnabledReturnsCorrectSchema()
+        {
+            // Arrange
+            IEdmModel model = EdmModelHelper.GraphBetaModel;
+            ODataContext context = new(model, new OpenApiConvertSettings
+            {
+                EnableDiscriminatorValue = true,
+            });
+
+            IEdmEntityType entity = model.SchemaElements.OfType<IEdmEntityType>().First(t => t.Name == "directoryObject");
+            Assert.NotNull(entity); // Guard
+
+            // Act
+            var schema = context.CreateStructuredTypeSchema(entity);
+            string json = schema.SerializeAsJson(OpenApiSpecVersion.OpenApi3_0);
+
+            // Assert
+            Assert.NotNull(json);
+            Assert.Equal(@"{
+  ""allOf"": [
+    {
+      ""$ref"": ""#/components/schemas/microsoft.graph.entity""
+    },
+    {
+      ""title"": ""directoryObject"",
+      ""type"": ""object"",
+      ""properties"": {
+        ""deletedDateTime"": {
+          ""pattern"": ""^[0-9]{4,}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])T([01][0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9]([.][0-9]{1,12})?(Z|[+-][0-9][0-9]:[0-9][0-9])$"",
+          ""type"": ""string"",
+          ""format"": ""date-time"",
+          ""nullable"": true
+        }
+      },
+      ""discriminator"": {
+        ""propertyName"": ""@odata.type"",
+        ""mapping"": {
+          ""microsoft.graph.directoryObject"": ""#/components/schemas/microsoft.graph.directoryObject""
```

Comment:
```suggestion
          ""#microsoft.graph.directoryObject"": ""#/components/schemas/microsoft.graph.directoryObject""
```

Shouldn't it be this (pound sign) according to odata conventions?

Reply:
Not according to this provided example:

![image](https://user-images.githubusercontent.com/40403681/152944536-02a6b8f6-201f-4b6b-894f-2d50134b9b6a.png)

@darrelmiller can help clarify.

Reply:
My reading of the example is consistent with @baywet 's suggestion.  The key of the mapping is the value that is present in the response payload.  In OData the @OData.Type property starts with a # character and therefore the mapping key should also.